src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/RuntimePlatformCheckAnalyzerTests.cs

```diff
@@ -0,0 +1,785 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.RuntimePlatformCheckAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices.UnitTests
+{
+    public class RuntimePlatformCheckAnalyzerTests
+    {
+        private readonly string PlatformCheckApiSource = @"
```

Comment:
@terrajobst @buyaa-n You may want to review the tests here, which are primarily added to validate and demonstrate the DFA support added here. It uses a dummy skeleton analyzer that just creates a diagnostic with a string representation of the available DFA state at all method invocations. This would be replaced be real analyzer checks for platform specific APIs + real diagnostics.